zebra-chain/src/work/difficulty.rs

```diff
@@ -123,6 +123,12 @@ impl fmt::Debug for ExpandedDifficulty {
 #[derive(Clone, Copy, Default, Eq, PartialEq, Ord, PartialOrd)]
 pub struct Work(u128);
 
+impl From<Work> for u128 {
+    fn from(work: Work) -> Self {
+        work.0
+    }
+}
```

Comment:
We only want to generate `Work` values from `ExpandedDifficulty` values in consensus-critical code.

So let's move this impl to a test or arbitrary module.

Reply:
We cant, if its in test code it won't be accessible from other crates, and the orphan rules prevent it from being defined in other crates. Also, this isn't constructing work, this was me trying to get access to the inner value, but tbh this shouldn't be done as a `From` impl, it should just be a method.

Reply:
Thanks, I missed that, too much code reading today :-)